include/laser_geometry/visibility_control.hpp

```diff
@@ -0,0 +1,49 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
```

Comment:
@jonbinney quick note: As all the code on this repo is BSD, you may prefer to have this header as BSD as well to avoid having different licenses on different files in the same package.

Reply:
@mikaelarguedas good point - but according to the apache license, it sounds like i can only modify the license for the modifications that i make, right? So since the original file from rclcpp is apache license, those portions of the file are always under apache license, aren't they?

Reply:
@jonbinney I can see that it has been relicensed to BSD already in some ROS 2 packages forked from ROS 1 (https://github.com/ros2/urdf/blob/ros2/urdf/include/urdf/visibility_control.hpp) so your should be fine.

Then for the origin of the code, talking with the original author of that file in rclcpp (:wink: @wjwwood), it looks like this code has actually been copied from a code in public domain.
I couldn't find the link to the original place it was copied from (before being copied to the gcc wiki) but my assumption is that we should be able to relicense it as BSD with permission of the original author of the rclcpp file.

Reply:
I can't find it either, but this is the origin of that code (near as I can tell):

http://www.nedprod.com/programs/gccvisibility.html

And the GCC wiki mentions this:

> > Note: the text on this page was almost integrally written by Niall Douglas, the original author of the patch, and placed on nedprod.com. This is basically a local mirror (especially useful because the external website now appears to be down).

> -- https://gcc.gnu.org/wiki/Visibility

Reply:
Makes sense. I'll update the license in a separate PR.